packages/vanilla/src/components/segmented-button-group/segmented-button-group.stories.js

```diff
@@ -1,32 +1,45 @@
 export default {
   title: 'Patterns/Segmented Button Group',
-  argTypes: {},
+  argTypes: {
+    selection: {
+      name: 'Selection',
+      description: 'Choose whether a single or multiple selections are allowed.',
+      control: 'radio',
+      options: ['single', 'multi']
+    },
+  },
 };
 
-const SingleSelectTemplate = () => {
+
+function generateButtons(buttons) {
+  let html=[];
+  buttons.forEach( ({ label, value, disabled }, i) => {
+    html=[...html, `<button type="button" class="cbp-btn" value="${value}" ${disabled==true ? 'disabled' : ''}>${label}</button>`];
+  });
+  return [...html].join("");
+};
```

Comment:
I think this function can be simplified with `.map()` and you won't have to use the spread syntax to push each template literal of the button into the `html` array. Instead of initializing a `let` with an empty array, you can do something like:
```javascript
function generateButtons(buttons) {
  const html = buttons.map(({ label, value, disabled}) => {
    return `<button type="button" class="cbp-btn" value="${value}" ${disabled==true ? 'disabled' : ''}>${label}  </button>`
  })

  return html.join("")
};
```

Reply:
Good call. I was struggling with making this work inline in the template literal and didn't really revisit it.